src/Core/src/Credentials/Credentials.php

```diff
@@ -62,4 +68,22 @@ public function getCredentials(Configuration $configuration): ?Credentials
     {
         return $this->isExpired() ? null : $this;
     }
+
+    public static function adjustExpireDate(?\DateTimeImmutable $expireDate, ?\DateTimeImmutable $reference = null): ?\DateTimeImmutable
```

Comment:
Should we force this to never return nullable?

Reply:
No, we cant, we are returning null if `$expireDate` is null.

Reply:
yes you're right.